ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/BerlinGasCalculator.java

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.mainnet;
+
+import static org.hyperledger.besu.ethereum.core.Address.BLS12_MAP_FP2_TO_G2;
+
+import org.hyperledger.besu.ethereum.core.Account;
+import org.hyperledger.besu.ethereum.core.Address;
+import org.hyperledger.besu.ethereum.core.Gas;
+import org.hyperledger.besu.ethereum.core.Wei;
+import org.hyperledger.besu.ethereum.vm.MessageFrame;
+
+import org.apache.tuweni.units.bigints.UInt256;
+
+public class BerlinGasCalculator extends IstanbulGasCalculator {
+
+  // new constants for EIP-2929
+  private static final Gas COLD_SLOAD_COST = Gas.of(2100);
+  private static final Gas COLD_ACCOUNT_ACCESS_COST = Gas.of(2600);
+  private static final Gas WARM_STORAGE_READ_COST = Gas.of(100);
+
+  // redefinitions for EIP-2929
+  private static final Gas SLOAD_GAS = WARM_STORAGE_READ_COST;
+  private static final Gas SSTORE_RESET_GAS = Gas.of(5000L).minus(COLD_SLOAD_COST);
+
+  // unchanged from Istanbul
+  private static final Gas SSTORE_SET_GAS = Gas.of(20_000);
+  private static final Gas SSTORE_CLEARS_SCHEDULE = Gas.of(15_000);
+
+  private static final Gas SSTORE_SET_GAS_LESS_SLOAD_GAS = SSTORE_SET_GAS.minus(SLOAD_GAS);
+  private static final Gas SSTORE_RESET_GAS_LESS_SLOAD_GAS = SSTORE_RESET_GAS.minus(SLOAD_GAS);
+  private static final Gas NEGATIVE_SSTORE_CLEARS_SCHEDULE = Gas.ZERO.minus(SSTORE_CLEARS_SCHEDULE);
+
+  // unchanged from Frontier
```

Comment:
dittto from above

Reply:
Again, I chose to redefine over escalating visibility to prevent future accidental shadowing. A form of this issue caused one of the Geth failures in early testing.